### PR TITLE
build: add useful workflows for maintenance

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
## Description
This PR adds useful GitHub workflows for an easier maintenance job:
- Pypi publishing after new release
- Python requirements update every Monday
- Test suite for better QA

## How to test
Either the python reqs and pypi publish job are used in Open edX plugins like: 
https://github.com/openedx/tutor-contrib-coursegraph/blob/master/.github/workflows/pypi-publish.yml
A similar version: https://github.com/openedx/openedx-events/blob/main/.github/workflows/pypi-publish.yml
ad they're are proven to be working:
https://pypi.org/project/tutor-contrib-coursegraph/
Now, for the test workflow, run the following commands locally:

```
make dev-requirements
make test-lint
make test-format
make test-pythonpackage
```

## References
Coursegraph plugin existent workflows: https://github.com/openedx/tutor-contrib-coursegraph/tree/master/.github/workflows
Community health files templates: https://github.com/openedx/.github

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by mariajgrimaldi** [comment]
note to author: fix failing tests

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.8
+
+      - name: Upgrade pip
+        run: python -m pip install --upgrade pip setuptools==44.0.0
```
Comment (bmtcril) [inline .github/workflows/test.yml:22]: Same question as the other PR about why we're pinning here instead of using pip.txt like the other actions

**Comment by itsjeyd** [comment]
label: core contributor

File: .github/workflows/test.yml (hunk 1 above)
Comment (mariajgrimaldi) [reply .github/workflows/test.yml]: Addressed!

**Comment by mariajgrimaldi** [comment]
@bmtcril @itsjeyd: can we remove this hardcoded todo comment? https://github.com/openedx/tutor-contrib-oars/actions/runs/4812683295/jobs/8568235934?pr=22#step:5:14

**Comment by bmtcril** [comment]
> @bmtcril @itsjeyd: can we remove this hardcoded todo comment? https://github.com/openedx/tutor-contrib-oars/actions/runs/4812683295/jobs/8568235934?pr=22#step:5:14

Yep, I'll put a task on the board to address it properly

**Comment by mariajgrimaldi** [comment]
there's another one here https://github.com/openedx/tutor-contrib-oars/blob/0019a992e333a5871e056c7f4968e6088aae5474/tutoroars/templates/oars/apps/pythonpath/superset-dashboard.py#L66, can we track it as well?

**Review by bmtcril — APPROVED** [review]
Looks good, thank you!

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/tutor-contrib-oars&entry.867055334=https://github.com/openedx/tutor-contrib-oars/pull/22&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2023-04-18+20:53&entry.1917517419=2023-04-27+13:01&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**